agent/setup.go

```diff
@@ -123,6 +126,10 @@ func NewBaseDeps(configLoader ConfigLoader, logOut io.Writer) (BaseDeps, error)
 	return d, nil
 }
 
+// grpcLogInitOnce because the test suite will call NewBaseDeps in many tests and
+// causes data races when it is re-initialized.
+var grpcLogInitOnce sync.Once
```

Comment:
Doesn't this mean that these logs are useless during tests?

Reply:
I guess it depends on how the logs are setup. If the logs are setup to go to stdout/stderr then it should work fine, they'll be attributed to the active test. If the logs are sent via `t.Log` it might hide some.

Although not doing this doesn't really fix the issue either, because the gRPC logger is a global. So if multiple tests are run in parallel, we would send all of the logs to one test logger, and none to the other.

This is only the logs for the grpc client and server though, so probably not important for any tests of Consul behaviour.